.github/workflows/docs-preview.yml

```diff
@@ -0,0 +1,140 @@
+        run: |
+          if [ ! -d ~/miniconda ]
+          then
+            wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O ~/miniconda.sh
+            bash ~/miniconda.sh -b -p ~/miniconda
+            rm ~/miniconda.sh
+          fi
+          export PATH="$HOME/miniconda/bin:$PATH"
+          conda --version
+      - name: Create conda environment
+        if: ${{ needs.should_run_docs.outputs.should_run_docs == 'true' }}
+        working-directory: tbp.monty
+        run: |
+          export PATH="$HOME/miniconda/bin:$PATH"
+          (conda env list | grep tbp.monty) && conda remove --name tbp.monty --all --yes || true
+          conda env create
+          source activate tbp.monty
+          pip install -e .[dev,github_readme_sync_tool,print_version_tool]
+      - name: Get version and branch
+        if: ${{ needs.should_run_docs.outputs.should_run_docs == 'true' }}
+        working-directory: tbp.monty
+        run: |
+          export PATH="$HOME/miniconda/bin:$PATH"
+          source activate tbp.monty
+          echo "VERSION=$(python -m tools.print_version.cli minor)" >> $GITHUB_ENV
+          # Make branch name safe for semver by replacing invalid chars with dash
+          # Remove trailing dash if present
+          echo "BRANCH_NAME=$(echo ${GITHUB_HEAD_REF} | tr -c '[:alnum:]' '-' | sed 's/-*$//')" >> $GITHUB_ENV
+          # Get current date in ISO format
+          echo "DATE=$(date -u +'%Y-%m-%d %H:%M UTC')" >> $GITHUB_ENV
+      - name: Deploy docs
+        if: ${{ needs.should_run_docs.outputs.should_run_docs == 'true' }}
+        working-directory: tbp.monty
+        run: |
+          export PATH="$HOME/miniconda/bin:$PATH"
+          source activate tbp.monty
+          export README_API_KEY=${{ secrets.README_API_KEY }}
+          export IMAGE_PATH=${{ vars.IMAGE_PATH }}
+          python -m tools.github_readme_sync.cli upload docs "${VERSION}-${BRANCH_NAME}"
+      - name: Update PR comment on success
+        if: ${{ success() && needs.should_run_docs.outputs.should_run_docs == 'true' }}
+        uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          message: |
+            📚 **Documentation Preview**
+
+            ✅ A preview of the documentation changes in this PR is available for maintainers at:
+            ${{ vars.DOCS_URL_PREFIX }}v${{ env.VERSION }}-${{ env.BRANCH_NAME }}
+
+            Last updated: ${{ env.DATE }}
+      - name: Update PR comment on failure
+        if: ${{ failure() && needs.should_run_docs.outputs.should_run_docs == 'true' }}
+        uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          message: |
+            📚 **Documentation Preview**
+
+            ❌ Failed to build documentation preview.
+
+            Check the [workflow logs](${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}) for details.
+
+  should_run_docs:
```

Comment:
**suggestion**: For non-required checks, we do not need to go through an artisanal setup of "should run," as seen in other workflows.

We can use the default paths filtering mechanism when a check is not required. See https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore.

The artisanal "should run" code is necessary elsewhere because when a workflow is skipped using the paths mechanism above, it registers as "skipped" instead of "success." However, a required check must "succeed" and cannot be "skipped."